### PR TITLE
Fix --package-type option

### PR DESCRIPTION
**Pull request by kylewillmon**
Fix the `--package-type` option by avoiding the call to `get_flag` (since this option is not a flag).

### CODE REVIEW

**Review by maxrake — DISMISSED** [review]
Confirmed with a local build and testing. LGTM.

File: cli/src/commands/packages.rs [hunk 1]
```diff
@@ -11,28 +12,28 @@ use crate::filter::{Filter, FilterIssues};
...
     // If a package type was provided on the command line, prefer that
-    //  to the global setting
-    if options.get_flag("package-type") {
-        package_type = PackageType::from_str(options.get_one::<String>("package-type").unwrap())
-            .unwrap_or(package_type);
-    }
+    // to the global setting
+    let package_type = match matches.get_one::<String>("package-type") {
+        Some(pt) => {
+            PackageType::from_str(pt).map_err(|()| anyhow!("invalid package type: {}", pt))?
```
Comment (cd-work) [inline cli/src/commands/packages.rs:24]:
```suggestion
            PackageType::from_str(pt).map_err(|_| anyhow!("invalid package type: {}", pt))?
```

**Review by cd-work — CHANGES_REQUESTED** [review]

File: cli/src/commands/packages.rs (hunk 1 above)
Comment (kylewillmon) [reply cli/src/commands/packages.rs:24]: My goal in using `()` was to highlight the fact that the error type is `()`. If it were any other reasonable error type, I wouldn't have to use `map_err`. And if we change the error type in the future (and we should), then we should reconsider the use of `map_err` here.

Comment (cd-work) [reply cli/src/commands/packages.rs:24]: I'm not sure anyone reading this in the future will get that from just looking at the `()`? Like I understand where you're coming from with the intention, but at the end of the day all this really does is reduce code consistency.

**Review by cd-work — APPROVED** [review]

**Review by maxrake — APPROVED** [review]

**Pull request merged**